platforms/nuttx/src/px4/common/px4_init.cpp

```diff
@@ -49,6 +49,103 @@
 # include <nuttx/i2c/i2c_master.h>
 #endif // CONFIG_I2C
 
+#if defined(GPIO_OTGFS_VBUS) && defined(CONFIG_SYSTEM_CDCACM)
+__BEGIN_DECLS
+#include <nuttx/wqueue.h>
+#include <builtin/builtin.h>
+
+extern int sercon_main(int c, char **argv);
+extern int serdis_main(int c, char **argv);
+__END_DECLS
+
+#include <uORB/Subscription.hpp>
+#include <uORB/topics/actuator_armed.h>
+
+static struct work_s usb_serial_work;
+static bool vbus_present_prev = false;
+
+enum class UsbAutoStartState {
+	disconnected,
+	connecting,
+	connected,
+	disconnecting,
+} usb_auto_start_state{UsbAutoStartState::disconnected};
+
+static void mavlink_usb_check(void *arg)
+{
+	int rescheduled = -1;
+
+	uORB::SubscriptionData<actuator_armed_s> actuator_armed_sub{ORB_ID(actuator_armed)};
+
+	const bool armed = actuator_armed_sub.get().armed;
+	const bool vbus_present = px4_arch_gpioread(GPIO_OTGFS_VBUS);
```

Comment:
This is not HW independent. 

replace with `board_read_VBUS_state`

Then where you added aliases to  GPIO_OTGFS_VBUS, remove them, that is not the correct way to do this. 


I.e 
```

return (getreg32(IMXRT_USB_ANALOG_USB1_VBUS_DETECT_STAT) & USB_ANALOG_USB_VBUS_DETECT_STAT_VBUS_VALID) ? 0 : 1;
```
}